mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java

```diff
@@ -72,7 +77,8 @@ public void delete(Mailbox mailbox) {
         Flux.merge(
                 mailboxPathDAO.delete(mailbox.generateAssociatedPath()),
                 mailboxPathV2DAO.delete(mailbox.generateAssociatedPath()))
-            .thenEmpty(mailboxDAO.delete(mailboxId))
+            .thenEmpty(Mono.defer(() -> mailboxDAO.delete(mailboxId))
```

Comment:
Why in the mapper and not in the dao?
Why for this specific request and not for all requests to Cassandra?

Reply:
It's related to the testing strategy...